src/shared/condition.c

```diff
@@ -262,6 +263,44 @@ static int condition_test_kernel_version(Condition *c, char **env) {
         return true;
 }
 
+static int condition_test_osrelease(Condition *c, char **env) {
+        _cleanup_free_ char *key = NULL, *actual_value = NULL;
+        const char *parameter = c->parameter, *requested_value;
+        OrderOperator order;
+        int r;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_OS_RELEASE);
+
+        r = extract_first_word(&parameter, &key, "!<=>", 0);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to parse parameter: %m");
+        if (r < 1)
+                return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Failed to parse parameter, key/value format expected: %m");
+
+        /* parse_order() needs the string to start with the comparators, so reset the variable */
+        parameter = c->parameter + strlen(key);
+        requested_value = strstrip((char *)parameter);
```

Comment:
this doesn't look right. You already skipped over the spaces before the comparison operator due to the way extract_first_word() is called (i.e. without whitespace as acknowledged separator), thus this strstrip() will only remove whitespace at the end of the whole expression, but that's weird, in particular as you are effectively changing the c->parameter field here, and I don't think testing a condition once should change the condition expression string for subsequent calls.

Reply:
I figure it would be good to figure out whether we want to allow extra whitespace before/after the comparison operator. I am leaning toward saying no to that, as env var files don#t allow that either before or after the "=" character.

Reply:
Whitespace not allowed anywmore before/after the key/separator